src/OmniSharp.Cake/Services/RequestHandlers/Diagnostics/CodeCheckHandler.cs

```diff
@@ -18,5 +22,33 @@ public class CodeCheckHandler : CakeRequestHandler<CodeCheckRequest, QuickFixRes
 
         public override bool IsValid(CodeCheckRequest request) =>
             !string.IsNullOrEmpty(request.FileName);
+
+        protected override Task<QuickFixResponse> TranslateResponse(QuickFixResponse response, CodeCheckRequest request)
+        {
+            if (response?.QuickFixes == null)
+            {
+                return Task.FromResult(response);
+            }
+
+            var quickFixes = response.QuickFixes.Where(x => PathsAreEqual(x.FileName, request.FileName));
+            response.QuickFixes = quickFixes;
+            return Task.FromResult(response);
+
+            bool PathsAreEqual(string x, string y)
+            {
+                if (x == null && y == null)
+                {
+                    return true;
+                }
+                if (x == null || y == null)
```

Comment:
Another tip I had from a contributor at corefx was that `x == null | y == null` performs better than the branch introduced by `||`. Not that this is performance-critical though.